eppo_core/src/events/event_dispatcher.rs

```diff
@@ -78,78 +76,96 @@ impl<T: EventQueue + Clone + Send + 'static> EventDispatcher<T> {
-                                },
-                                Some(EventDispatcherCommand::Event) => {
-                                    if event_queue.is_batch_full() {
-                                        // Event queue batch is full, break loop and deliver it
-                                        break;
-                                    } // else loop to get more events
-                                },
-                                Some(EventDispatcherCommand::Flush) => {
-                                    break;
-                                }
-                            }
-                        }
                     }
                 }
             }
+        }
+    }
+}
 
-            let event_delivery = event_delivery.clone();
-            let batch = event_queue.next_batch(QueuedEventStatus::Pending);
-            let queue = event_queue.clone();
-            // Send events
-            tokio::spawn({
-                async move {
-                    // Spawning a new task, so the main task can continue batching events and respond to
-                    // commands. At this point, batch_queue is guaranteed to have at least one event.
-                    let events = batch
-                        .iter()
-                        .map(|queued_event| queued_event.clone().event)
-                        .collect();
-                    let result = event_delivery.deliver(events).await;
-                    match result {
-                        Ok(response) => {
-                            let failed_event_uuids = response.failed_events;
-                            if !failed_event_uuids.is_empty() {
-                                warn!("Failed to deliver {} events", failed_event_uuids.len());
-                                let failed_events = batch
-                                    .into_iter()
-                                    .filter(|queued_event| {
-                                        failed_event_uuids.contains(&queued_event.event.uuid)
-                                    })
-                                    .collect();
-                                queue.mark_events_as_failed(failed_events);
-                            }
-                        }
-                        Err(err) => {
-                            warn!("Failed to deliver events: {}", err);
-                            queue.mark_events_as_failed(batch);
-                        }
+fn spawn_event_delivery<T: EventQueue + Clone + Send + 'static>(
+    batch: Vec<QueuedEvent>,
+    event_delivery: EventDelivery,
+    queue: T,
+) {
+    if batch.is_empty() {
+        // nothing to deliver
+        return;
+    }
+    // Send events
+    tokio::spawn({
+        async move {
+            // Spawning a new task, so the main task can continue batching events and respond to
+            // commands. At this point, batch is guaranteed to have at least one event.
```

Comment:
We have agreed to make delivery "single-threaded" for the first version, so we can just drop `spawn` and do everything inline.

Otherwise, we don't have a concurrency limit, so it may cause unlimited number of concurrent http requests (maybe still limited by reqwest client but probably too high)